File: paykit-mobile/kotlin/BitkitUI/NavigationExample.kt
package com.paykit.mobile.bitkit

import androidx.compose.foundation.layout.padding
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.navigation.compose.NavHost
import androidx.navigation.compose.composable
import androidx.navigation.compose.rememberNavController
import com.paykit.mobile.paykit_mobile.PaykitClient

/**
 * Navigation destinations for Bitkit
 */
sealed class BitkitScreen(val route: String, val title: String, val icon: @Composable () -> Unit) {
    object Dashboard : BitkitScreen(
        "dashboard",
        "Dashboard",
        { Icon(Icons.Default.Home, contentDescription = "Dashboard") }
    )
    object Send : BitkitScreen(
        "send",
        "Send",
        { Icon(Icons.Default.Send, contentDescription = "Send") }
    )
    object Receive : BitkitScreen(
        "receive",
        "Receive",
        { Icon(Icons.Default.CallReceived, contentDescription = "Receive") }
    )
    object Contacts : BitkitScreen(
        "contacts",
        "Contacts",
        { Icon(Icons.Default.Person, contentDescription = "Contacts") }
    )
    object Receipts : BitkitScreen(
        "receipts",
        "Receipts",
        { Icon(Icons.Default.ReceiptLong, contentDescription = "Receipts") }
    )
}

/**
 * Main navigation example for Bitkit
 * Bitkit should adapt this to their navigation structure
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun BitkitMainNavigation(
    paykitClient: PaykitClient,
    contactStorage: ContactStorageProtocol,
    receiptStorage: ReceiptStorageProtocol
) {
    val navController = rememberNavController()
    val selectedIndex = remember { mutableIntStateOf(0) }
    
    // Initialize ViewModels
    val dashboardViewModel = remember {
        BitkitDashboardViewModel(paykitClient)
    }
    val paymentViewModel = remember {
        BitkitPaymentViewModel(paykitClient)
    }
    val contactsViewModel = remember {
        BitkitContactsViewModel(contactStorage)
    }
    val receiptsViewModel = remember {
        BitkitReceiptsViewModel(receiptStorage)
    }
    
    Scaffold(
        bottomBar = {
            NavigationBar {
                BitkitScreen.values().forEachIndexed { index, screen ->
                    NavigationBarItem(
                        icon = screen.icon(),
                        label = { Text(screen.title) },
                        selected = selectedIndex.intValue == index,
                        onClick = {
                            selectedIndex.intValue = index
                            navController.navigate(screen.route) {
                                popUpTo(navController.graph.startDestinationId)
                                launchSingleTop = true
                            }
                        }
                    )
                }
            }
        }
    ) { padding ->
        NavHost(
            navController = navController,
            startDestination = BitkitScreen.Dashboard.route,
            modifier = Modifier.padding(padding)
        ) {
            composable(BitkitScreen.Dashboard.route) {
                BitkitDashboardScreen(
                    viewModel = dashboardViewModel,
                    onSendPayment = { navController.navigate(BitkitScreen.Send.route) },
                    onReceivePayment = { navController.navigate(BitkitScreen.Receive.route) },
                    onViewReceipts = { navController.navigate(BitkitScreen.Receipts.route) },
                    onViewContacts = { navController.navigate(BitkitScreen.Contacts.route) }
                )
            }
            composable(BitkitScreen.Send.route) {
                BitkitPaymentScreen(
                    viewModel = paymentViewModel,
                    onPaymentComplete = { navController.navigate(BitkitScreen.Receipts.route) }
                )
            }
            composable(BitkitScreen.Receive.route) {
                // Receive screen - Bitkit should implement
                Text("Receive Payment")
            }
            composable(BitkitScreen.Contacts.route) {
                BitkitContactsScreen(viewModel = contactsViewModel)
            }
            composable(BitkitScreen.Receipts.route) {
                BitkitReceiptsScreen(viewModel = receiptsViewModel)
            }
        }
    }
    
    // Load initial data
    LaunchedEffect(Unit) {
        dashboardViewModel.loadDashboard(
            receiptStorage = receiptStorage,
            contactStorage = contactStorage
        )
    }
}

// Extension to get all screens
private fun BitkitScreen.values(): List<BitkitScreen> {
    return listOf(
        BitkitScreen.Dashboard,
        BitkitScreen.Send,
        BitkitScreen.Receive,
        BitkitScreen.Contacts,
        BitkitScreen.Receipts
    )
}

File: paykit-mobile/swift/BitkitUI/NavigationExample.swift
//
//  NavigationExample.swift
//  PaykitMobile
//
//  Navigation structure example for Bitkit integration.
//  This shows how to set up the main navigation with all screens.
//

import SwiftUI
import PaykitMobile

/// Main navigation view for Bitkit
/// Bitkit should adapt this to their navigation structure
public struct BitkitMainNavigationView: View {
    @StateObject private var paykitClient = PaykitClient.new()
    
    // View Models
    @StateObject private var dashboardViewModel: BitkitDashboardViewModel
    @StateObject private var paymentViewModel: BitkitPaymentViewModel
    @StateObject private var receiveViewModel: BitkitReceivePaymentViewModel
    @StateObject private var contactsViewModel: BitkitContactsViewModel
    @StateObject private var receiptsViewModel: BitkitReceiptsViewModel
    @StateObject private var paymentMethodsViewModel: BitkitPaymentMethodsViewModel
    
    @State private var selectedTab = 0
    
    public init() {
        let client = PaykitClient.new()
        
        // Initialize view models
        // Bitkit should provide their storage implementations
        _dashboardViewModel = StateObject(wrappedValue: BitkitDashboardViewModel(paykitClient: client))
        _paymentViewModel = StateObject(wrappedValue: BitkitPaymentViewModel(paykitClient: client))
        _receiveViewModel = StateObject(wrappedValue: BitkitReceivePaymentViewModel(paykitClient: client))
        
        // Placeholder storage - Bitkit should replace with actual implementations
        let contactStorage = MockContactStorage()
        let receiptStorage = MockReceiptStorage()
        
        _contactsViewModel = StateObject(wrappedValue: BitkitContactsViewModel(contactStorage: contactStorage))
        _receiptsViewModel = StateObject(wrappedValue: BitkitReceiptsViewModel(receiptStorage: receiptStorage))
        _paymentMethodsViewModel = StateObject(wrappedValue: BitkitPaymentMethodsViewModel(paykitClient: client))
    }
    
    public var body: some View {
        TabView(selection: $selectedTab) {
            // Dashboard Tab
            BitkitDashboardView(viewModel: dashboardViewModel)
                .tabItem {
                    Label("Dashboard", systemImage: "house.fill")
                }
                .tag(0)
            
            // Send Tab
            BitkitPaymentView(viewModel: paymentViewModel)
                .tabItem {
                    Label("Send", systemImage: "arrow.up.circle.fill")
                }
                .tag(1)
            
            // Receive Tab
            BitkitReceivePaymentView(viewModel: receiveViewModel)
                .tabItem {
                    Label("Receive", systemImage: "arrow.down.circle.fill")
                }
                .tag(2)
            
            // Contacts Tab
            BitkitContactsView(viewModel: contactsViewModel)
                .tabItem {
                    Label("Contacts", systemImage: "person.2.fill")
                }
                .tag(3)
            
            // Receipts Tab
            BitkitReceiptsView(viewModel: receiptsViewModel)
                .tabItem {
                    Label("Receipts", systemImage: "doc.text.fill")
                }
                .tag(4)
        }
        .onAppear {
            // Load initial data
            dashboardViewModel.loadDashboard()
        }
    }
}

// MARK: - Mock Storage (for example only - Bitkit should implement real storage)

class MockContactStorage: ContactStorageProtocol {
    func listContacts() -> [Contact] {
        return []
    }
}

class MockReceiptStorage: ReceiptStorageProtocol {
    func recentReceipts(limit: Int) -> [Receipt] {
        return []
    }
    
    func totalSent() -> UInt64 {
        return 0
    }
    
    func totalReceived() -> UInt64 {
        return 0
    }
    
    func pendingCount() -> Int {
        return 0
    }
}
